float64 a0
float64 a1
float64 a2
float64 a3
float64 t0
float64 tf
